### PR TITLE
Add checksum for py-requests 2.27.1, 2.28.0, 2.28.1

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-requests/package.py [hunk 1]
```diff
@@ -27,7 +30,8 @@ class PyRequests(PythonPackage):
     variant("socks", default=False, description="SOCKS and HTTP proxy support")
 
     depends_on("python@2.7:2.8,3.5:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.6:", type=("build", "run"), when="@2.26.0:")
+    depends_on("python@2.7:2.8,3.6:", type=("build", "run"), when="@2.26.0:2.27")
+    depends_on("python@3.7:", type=("build", "run"), when='@2.28.0:")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-requests/package.py]:
```suggestion
    depends_on("python@3.7:", type=("build", "run"), when="@2.28.0:")
```

**Review by alalazo — DISMISSED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Comment by iarspider** [comment]
@spackbot run pipeline

No idea why it failed - the log is too long and thus truncated

**Comment by iarspider** [comment]
@adamjstewart  I have tried reproducing the failure locally: `$ spack install ecp-data-vis-sdk ~cuda ~rocm ~sensei +adios2 +ascent +cinema +darshan +faodel +hdf5 +paraview +pnetcdf +sz +unifyfs +veloc +vtkm +zfp +visit`

The failing package is `llvm`, which can't find some nCurses symbols at link-time:
```
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Window::~Window()':
IOHandlerCursesGUI.cpp:(.text._ZN6curses6WindowD0Ev[_ZN6curses6WindowD5Ev]+0x443): undefined reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `std::_Sp_counted_ptr_inplace<curses::Window, std::allocator<curses::Window>, (__gnu_cxx::_Lock_policy)2>::_M_disp
ose()':
IOHandlerCursesGUI.cpp:(.text._ZNSt23_Sp_counted_ptr_inplaceIN6curses6WindowESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt23_Sp_counted_ptr_inplaceIN6curses6WindowESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0x3d3): undefined reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Window::~Window()':
IOHandlerCursesGUI.cpp:(.text._ZN6curses6WindowD2Ev[_ZN6curses6WindowD5Ev]+0x433): undefined reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Menu::DrawMenuTitle(curses::Window&, bool)':
IOHandlerCursesGUI.cpp:(.text._ZN6curses4Menu13DrawMenuTitleERNS_6WindowEb+0x172): undefined reference to `acs_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Menu::WindowDelegateDraw(curses::Window&, bool)':
IOHandlerCursesGUI.cpp:(.text._ZN6curses4Menu18WindowDelegateDrawERNS_6WindowEb+0x53): undefined reference to `acs_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `TreeItem::DrawTreeForChild(curses::Window&, TreeItem*, unsigned int)':
IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem16DrawTreeForChildERN6curses6WindowEPS_j[_ZN8TreeItem16DrawTreeForChildERN6curses6WindowEPS_j]+0x42): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem16DrawTreeForChildERN6curses6WindowEPS_j[_ZN8TreeItem16DrawTreeForChildERN6curses6WindowEPS_j]+0x73): undefined reference to `acs
_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem16DrawTreeForChildERN6curses6WindowEPS_j[_ZN8TreeItem16DrawTreeForChildERN6curses6WindowEPS_j]+0xa8): undefined reference to `acs
_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Application::TerminalSizeChanged()':
IOHandlerCursesGUI.cpp:(.text._ZN6curses11Application19TerminalSizeChangedEv[_ZN6curses11Application19TerminalSizeChangedEv]+0x51d): undefined reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `TreeItem::Draw(curses::Window&, int, unsigned int, int&, int&)':
IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem4DrawERN6curses6WindowEijRiS3_[_ZN8TreeItem4DrawERN6curses6WindowEijRiS3_]+0xf2): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem4DrawERN6curses6WindowEijRiS3_[_ZN8TreeItem4DrawERN6curses6WindowEijRiS3_]+0x12b): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem4DrawERN6curses6WindowEijRiS3_[_ZN8TreeItem4DrawERN6curses6WindowEijRiS3_]+0x1b3): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem4DrawERN6curses6WindowEijRiS3_[_ZN8TreeItem4DrawERN6curses6WindowEijRiS3_]+0x1c9): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN8TreeItem4DrawERN6curses6WindowEijRiS3_[_ZN8TreeItem4DrawERN6curses6WindowEijRiS3_]+0x24f): undefined reference to `acs_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Application::Run(lldb_private::Debugger&)':
IOHandlerCursesGUI.cpp:(.text._ZN6curses11Application3RunERN12lldb_private8DebuggerE[_ZN6curses11Application3RunERN12lldb_private8DebuggerE]+0x1a): undefined reference to `halfdelay'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Window::RemoveSubWindow(curses::Window*)':
IOHandlerCursesGUI.cpp:(.text._ZN6curses6Window15RemoveSubWindowEPS0_[_ZN6curses6Window15RemoveSubWindowEPS0_]+0x243): undefined reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Window::DrawTitleBox(char const*, char const*)':
IOHandlerCursesGUI.cpp:(.text._ZN6curses6Window12DrawTitleBoxEPKcS2_[_ZN6curses6Window12DrawTitleBoxEPKcS2_]+0xc3): undefined reference to `acs_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `Row::DrawTreeForChild(curses::Window&, Row*, unsigned int)':
IOHandlerCursesGUI.cpp:(.text._ZN3Row16DrawTreeForChildERN6curses6WindowEPS_j[_ZN3Row16DrawTreeForChildERN6curses6WindowEPS_j]+0x4b): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN3Row16DrawTreeForChildERN6curses6WindowEPS_j[_ZN3Row16DrawTreeForChildERN6curses6WindowEPS_j]+0x7b): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN3Row16DrawTreeForChildERN6curses6WindowEPS_j[_ZN3Row16DrawTreeForChildERN6curses6WindowEPS_j]+0xb0): undefined reference to `acs_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Application::GetMainWindow()':
IOHandlerCursesGUI.cpp:(.text._ZN6curses11Application13GetMainWindowEv[_ZN6curses11Application13GetMainWindowEv]+0x78): undefined reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `ValueObjectListDelegate::DisplayRows(curses::Window&, std::vector<Row, std::allocator<Row> >&, DisplayOptions&)':
IOHandlerCursesGUI.cpp:(.text._ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions[_ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions]+0x284): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions[_ZN23ValueObjectListDelegate11DisplayRows
ERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions]+0x343): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions[_ZN23ValueObjectListDelegate11DisplayRows
ERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions]+0x4a3): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions[_ZN23ValueObjectListDelegate11DisplayRows
ERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions]+0x4e3): undefined reference to `acs_map'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions[_ZN23ValueObjectListDelegate11DisplayRows
ERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions]+0x6eb): undefined reference to `acs_map'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o):IOHandlerCursesGUI.cpp:(.text._ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptio
ns[_ZN23ValueObjectListDelegate11DisplayRowsERN6curses6WindowERSt6vectorI3RowSaIS4_EER14DisplayOptions]+0x78b): more undefined references to `acs_map' follow
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `ApplicationDelegate::MenuDelegateAction(curses::Menu&)':
IOHandlerCursesGUI.cpp:(.text._ZN19ApplicationDelegate18MenuDelegateActionERN6curses4MenuE[_ZN19ApplicationDelegate18MenuDelegateActionERN6curses4MenuE]+0x5d7): undefined reference to `stdscr'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN19ApplicationDelegate18MenuDelegateActionERN6curses4MenuE[_ZN19ApplicationDelegate18MenuDelegateActionERN6curses4MenuE]+0x6d6): undefined 
reference to `stdscr'
(...)/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `lldb_private::IOHandlerCursesGUI::Activate()':
IOHandlerCursesGUI.cpp:(.text._ZN12lldb_private18IOHandlerCursesGUI8ActivateEv+0xb0f): undefined reference to `curs_set'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN12lldb_private18IOHandlerCursesGUI8ActivateEv+0xb1b): undefined reference to `stdscr'
(...)/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN12lldb_private18IOHandlerCursesGUI8ActivateEv+0xb28): undefined reference to `keypad'
```

**Comment by iarspider** [comment]
Full build log is here: https://gist.github.com/iarspider/60b9394d0d5939422a7ef80d15903f15

**Comment by adamjstewart** [comment]
I think you can ignore the CI failure, seems unrelated to this PR.

@scottwittenburg can you investigate this?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**